src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/OverseasEntitiesController.java

```diff
@@ -187,37 +198,45 @@ public ResponseEntity<Object> createNewSubmissionForSaveAndResume(
         ApiLogger.debugContext(requestId, "Called getValidationStatus(...)");
 
         var logMap = new HashMap<String, Object>();
         logMap.put(OVERSEAS_ENTITY_ID_KEY, submissionId);
         logMap.put(TRANSACTION_ID_KEY, transactionId);
 
-        ApiLogger.infoContext(requestId, "Calling service to get the overseas entity submission", logMap);
-        var submissionDtoOptional = overseasEntitiesService.getOverseasEntitySubmission(submissionId);
-        if (submissionDtoOptional.isPresent()) {
-            var validationStatus = new ValidationStatusResponse();
-            validationStatus.setValid(true);
+        try {
+            ApiLogger.infoContext(requestId, "Calling service to get the overseas entity submission", logMap);
+            var submissionDtoOptional = overseasEntitiesService.getOverseasEntitySubmission(submissionId);
+            if (submissionDtoOptional.isPresent()) {
+                var validationStatus = new ValidationStatusResponse();
+                validationStatus.setValid(true);
+
+                if (isValidationEnabled) {
+                    String passThroughTokenHeader = request.getHeader(ApiSdkManager.getEricPassthroughTokenHeader());
 
-            if (isValidationEnabled) {
-                var validationErrors = overseasEntitySubmissionDtoValidator.validateFull(
-                        submissionDtoOptional.get(), new Errors(), requestId);
+                    var validationErrors = overseasEntitySubmissionDtoValidator.validateFull(
+                            submissionDtoOptional.get(), new Errors(), requestId, passThroughTokenHeader);
 
-                if (validationErrors.hasErrors()) {
-                    final var errorsAsJsonString = convertErrorsToJsonString(validationErrors);
-                    ApiLogger.errorContext(requestId, String.format(VALIDATION_ERRORS_MESSAGE, errorsAsJsonString), null, logMap);
+                    if (validationErrors.hasErrors()) {
+                        final var errorsAsJsonString = convertErrorsToJsonString(validationErrors);
+                        ApiLogger.errorContext(requestId, String.format(VALIDATION_ERRORS_MESSAGE, errorsAsJsonString), null, logMap);
 
-                    flagValidationStatusAsFailed(validationStatus, errorsAsJsonString);
+                        flagValidationStatusAsFailed(validationStatus, errorsAsJsonString);
+                    }
                 }
+
+                return ResponseEntity.ok().body(validationStatus);
             }
 
-            return ResponseEntity.ok().body(validationStatus);
+            final var message = String.format("Could not find submission data for submission %s", submissionId);
+            ApiLogger.errorContext(requestId, message, null, logMap);
+            return ResponseEntity.notFound().build();
+        } catch (ServiceException e) {
+            ApiLogger.errorContext(requestId, "Error validating the Overseas Entity Submission", e, logMap);
```

Comment:
the error might also happen during the get of the overseas entities (inside the try) so don't think we can be certain the exception is caused by the validation. maybe change message to say something went wrong in getValidationStatus

Reply:
ignore that, the method is called getValidationStatus so it will be validating, message is ok.